VMEncryption/main/check_util.py

```diff
@@ -90,7 +94,83 @@ def is_invalid_lvm_os(self):
                     detected = True
         return detected
 
-    def is_precheck_failure(self):
+    def check_kv_url(self, test_url, message):
+        """basic sanity check of the key vault url"""
+        expected = "https://{keyvault-name}.vault.azure.net/"
+
+        if test_url is None:
+            raise Exception(message + '\nNo URL supplied')
+
+        try:
+            parse_result = urlparse.urlparse(test_url)
+        except:
+            raise Exception(message + '\nMalformed URL: ' + test_url)
+
+        if not (parse_result.scheme.lower() == "https" and \
+                re.match('[a-zA-Z0-9\-]+\.vault\.azure\.net(:443)?$', parse_result.netloc)):
+            raise Exception('\n' + message + '\nActual: ' + test_url + '\nExpected: ' + expected + "\n")
+        return
+
+    def check_kv_id(self, test_id, message):
+        """basic sanity check of the key vault id"""
+        # more strict checking would validate the full key vault id format
+        expected = "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.KeyVault/vaults/{vaultname}"
+
+        if test_id is None:
+            raise Exception(message + '\nNo Resource ID supplied')
+
+        id_splits = test_id.lower().split('/')
+
+        if not (len(id_splits) >= 9 and \
+                id_splits[0] == "" and \
```

Comment:
this only checks for non-empty, it does not validate that various naming conventions are being met 
https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions

Reply:
Taking a look

Reply:
I don't feel this needs to hold up the rest of the PR, it might be something to note and then pick up in a future round of changes.

Reply:
I agree, if it looks like people are really prone to making a key vault id error we can look into this more